mautic/mautic
# DPMMA-1696 optionally stop engaging after closing the focus

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ N ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds a new option to the focus item builder to stop engaging when the user directly closes the focus (using the x key).
The implementation is similar to the existing **Stop engaging after a conversion** and is based on cookies. Therefore, the focus will no longer be displayed until the visitor clears their cookies.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create new Focus Item with selected option **Stop engaging after a closing the focus?**
![image](https://github.com/mautic/mautic/assets/8580942/f9d19b05-4f3d-491b-92c5-31ce8f24e210)
3. Create Landing Page and add Focus Item (via script in head/footer or token in builder)
4. Open the landing page in a private window, the Focus Item should appear
5. Refresh the page without closing the Focus Item, it should appear again
6. Close the Focus Item with the 'x' button
7. Refresh the page
8. The Focus item should not appear until you clear your cookies
